test/functional/feature_utxo_set_hash.py

```diff
@@ -29,39 +34,29 @@ def test_muhash_implementation(self):
         mocktime = node.getblockheader(node.getblockhash(0))['time'] + 1
         node.setmocktime(mocktime)
 
-        # Generate 100 blocks and remove the first since we plan to spend its
-        # coinbase
-        block_hashes = self.generate(wallet, 1) + self.generate(node, 99)
-        blocks = list(map(lambda block: from_hex(CBlock(), node.getblock(block, False)), block_hashes))
-        blocks.pop(0)
-
-        # Create a spending transaction and mine a block which includes it
+        # Generate 100 blocks, create spending transaction and mine another block
+        self.generate(wallet, 1) + self.generate(node, 99)
         txid = wallet.send_self_transfer(from_node=node)['txid']
-        tx_block = self.generateblock(node, output=wallet.get_address(), transactions=[txid])
-        blocks.append(from_hex(CBlock(), node.getblock(tx_block['hash'], False)))
+        self.generateblock(node, output=wallet.get_address(), transactions=[txid])
 
-        # Serialize the outputs that should be in the UTXO set and add them to
-        # a MuHash object
+        # Serialize created/spent transaction outputs in each block and
```

Comment:
nit to make the link with the next sentence a bit more clear
```suggestion
        # Serialize (created/spent) transaction outputs in each block and
```

Reply:
Thanks, done.